include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -60,231 +70,104 @@ typename MatType::elem_type CMAES<SelectionPolicyType>::Optimize(
-
-  const double c1 = 2 / (std::pow(iterate.n_elem + 1.3, 2) + muEffective);
-  const double alphaMu = 2;
-  const double cmu = std::min(1 - c1, alphaMu * (muEffective - 2 + 1 /
-      muEffective) / (std::pow(iterate.n_elem + 2, 2) +
-      alphaMu * muEffective / 2));
-
-  std::vector<BaseMatType> mPosition(2, BaseMatType(iterate.n_rows,
-      iterate.n_cols));
-  mPosition[0] = lowerBound + arma::randu<BaseMatType>(
+  BaseMatType mCandidate(iterate.n_rows, iterate.n_cols);
+  
+  mCandidate = lowerBound + arma::randu<BaseMatType>(
       iterate.n_rows, iterate.n_cols) * (upperBound - lowerBound);
 
-  BaseMatType step(iterate.n_rows, iterate.n_cols);
-  step.zeros();
-
   // Calculate the first objective function.
+  size_t numFunctions = function.NumFunctions();
   ElemType currentObjective = 0;
   for (size_t f = 0; f < numFunctions; f += batchSize)
   {
     const size_t effectiveBatchSize = std::min(batchSize, numFunctions - f);
-    const ElemType objective = function.Evaluate(mPosition[0], f,
+    const ElemType objective = function.Evaluate(mCandidate, f,
         effectiveBatchSize);
     currentObjective += objective;
 
-    Callback::Evaluate(*this, function, mPosition[0], objective,
+    Callback::Evaluate(*this, function, mCandidate, objective,
         callbacks...);
   }
 
   ElemType overallObjective = currentObjective;
   ElemType lastObjective = std::numeric_limits<ElemType>::max();
 
   // Population parameters.
-  std::vector<BaseMatType> pStep(lambda, BaseMatType(iterate.n_rows,
-      iterate.n_cols));
-  std::vector<BaseMatType> pPosition(lambda, BaseMatType(iterate.n_rows,
-      iterate.n_cols));
+  std::vector<BaseMatType> z(lambda, BaseMatType(iterate.n_rows, iterate.n_cols));
+  std::vector<BaseMatType> y(lambda, BaseMatType(iterate.n_rows, iterate.n_cols));
+  // candidates is vector of x_i(g) with x_i(g) 
+  // is either a column or row vector but we generalize the type of data here)
+  std::vector<BaseMatType> candidates(lambda, BaseMatType(iterate.n_rows, iterate.n_cols)); 
   BaseMatType pObjective(lambda, 1); // pObjective is vector-shaped.
-  std::vector<BaseMatType> ps(2, BaseMatType(iterate.n_rows, iterate.n_cols));
-  ps[0].zeros();
-  ps[1].zeros();
-  std::vector<BaseMatType> pc = ps;
-  std::vector<BaseMatType> C(2, BaseMatType(iterate.n_elem, iterate.n_elem));
-  C[0].eye();
-
-  // Covariance matrix parameters.
-  arma::Col<ElemType> eigval; // TODO: might need a more general type.
-  BaseMatType eigvec;
-  BaseMatType eigvalZero(iterate.n_elem, 1); // eigvalZero is vector-shaped.
-  eigvalZero.zeros();
 
+  BaseMatType ps(iterate.n_rows, iterate.n_cols);
+  ps.zeros();
+  BaseMatType pc = ps;
+  // Sep and Vd update parameters
+  BaseMatType sepCov(iterate.n_rows, iterate.n_cols, arma::fill::ones);
+  BaseMatType sepCovinv = sepCov;
+  // v ~ N(0, I/d)
+  BaseMatType v = arma::randn<BaseMatType>(iterate.n_rows, iterate.n_cols) / std::sqrt(iterate.n_elem);
```

Comment:
```suggestion
  BaseMatType v = arma::randn<BaseMatType>(iterate.n_rows,
      iterate.n_cols) / std::sqrt(iterate.n_elem);
```

See https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping for more details.